lib/elixir/lib/kernel.ex

```diff
@@ -2642,6 +2642,179 @@ defmodule Kernel do
     end
   end
 
+  @doc """
+  Evaluates the expression corresponding to the first clause that
+  evaluates to a truthy value.
+
+      cond do
+        hd([1, 2, 3]) ->
+          "1 is considered as true"
+      end
+      #=> "1 is considered as true"
+
+  Raises an error if all conditions evaluate to `nil` or `false`.
+  For this reason, it may be necessary to add a final always-truthy condition
+  (anything non-`false` and non-`nil`), which will always match.
+
+  ## Examples
+
+      cond do
+        1 + 1 == 1 ->
+          "This will never match"
+        2 * 2 != 4 ->
+          "Nor this"
+        true ->
+          "This will"
+      end
+      #=> "This will"
+
+  """
+
+  defmacro cond(clauses) do
+    build_cond(clauses, __CALLER__)
+  end
+
+  defp build_cond([do: do_clause], caller) do
+    %{module: module, line: line} = caller
+
+    [{:->, meta, [[condition], body]} | tail] =
+      reversed_clauses = reverse_cond_clauses(do_clause, [])
+
+    case_clause =
+      case condition do
+        value when is_atom(value) and value != false and value != nil ->
+          build_cond_clauses(tail, body, meta, module)
+
+        _ ->
+          error =
+            quote do
+              :erlang.error(CondClauseError.exception([]))
+            end
+
+          build_cond_clauses(reversed_clauses, error, meta, module)
+      end
+
+    replace_case_meta(case_clause, line)
+  end
+
+  defp build_cond([], _caller) do
+    :erlang.error(ArgumentError.exception("missing :do option in \"cond\""))
+  end
+
+  defp build_cond(clauses, _caller) when is_list(clauses) do
+    check_at_most_once(:do, clauses, 0, "cond")
+    check_unexpected_options([:do], clauses, "cond")
+  end
+
+  defp build_cond(_invalid_args, _caller) do
+    invalid_arguments_error("cond")
+  end
+
+  defp build_cond_clauses([], acc, _old_meta, _module) do
+    acc
+  end
+
+  defp build_cond_clauses([{:->, new_meta, [[condition], body]} | tail], acc, old_meta, module) do
+    var = {:cond, [], module}
+
+    case_clause =
+      {:case, new_meta,
```

Comment:
I can't find a way to make this look nice

Reply:
Take a look at how `if`, `||` and `&&` are implemented. The principle here should be similar.